src/findings_data_import.py

```diff
@@ -49,12 +49,292 @@ def move_processed_file(s3_client, bucket, folder, filename):
+    """
+    Validate and return cleaned/processed finding.
+
+    Parameters
+    ----------
+    findings_data : dict
+        The findings dictionary pulled from a findings JSON file
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+
+    """
+    valid_findings = []
+
+    # if we (v2) get a lone object, wrap it in a list for compatability
+    if type(findings_data) != list:
+        findings_data = [findings_data]
+
+    # Iterate through data and save each record to the database
+    for index, finding in enumerate(findings_data):
+
+        if not finding or not hasattr(finding, "keys"):
+            logging.warning("Received an empty of invalid finding object, skipping.")
+            continue
+
+        # Replace or rename fields from replacement JSON
+        for field in field_map_dict:
+            if field in finding.keys():
+                if field_map_dict[field]:
+                    finding[field_map_dict[field]] = finding[field]
+                finding.pop(field, None)
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if "RVA ID" not in finding.keys() or (
+            not ("NCATS ID" in finding.keys() and "Severity" in finding.keys())
+            and "findings" not in finding.keys()
+        ):
+            logging.warning(
+                "Skipping record %d. Missing 'RVA ID' or 'NCATS ID' field.", index
+            )
+            continue
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            finding["RVA ID"] = rID
+        else:
+            logging.warning(
+                "Skipping record %d: Unable to extract valid RVA ID from '%s'",
+                index,
+                finding["RVA ID"],
+            )
+            continue
+
+        valid_findings.append(finding)
+
+    logging.info(
+        "{%d/%d} documents successfully processed.",
+        len(valid_findings),
+        len(findings_data),
     )
-    logging.warning(log_format, index, file, *args)
+
+    return valid_findings
+
+
+def update_record(db=None, finding=None):
+    """Insert or update a record, based on the (naively) detected schema type.
+
+    Parameters
+    ----------
+    db : MongoClient database
+        The database to update
+
+    finding: dict
+        The finding data to insert.
+    """
+    if "RVA ID" not in finding:
+        raise ValueError("The passed finding had no RVA ID field.")
+
+    # if it has "NCATS ID", it is 'v1' record
+    if "NCATS ID" in finding and "Severity" in finding:
+        finding["schema"] = "v1"
```

Comment:
Since the "v1" keys are all capitalized as far as I know I would prefer to be consistent.

```suggestion
        finding["Schema"] = "v1"
```